### ISSUE TITLE
`afterRender` hook called multiple times when there are multiple root views in the application

### ISSUE THREAD

**Issue by atscott** [angular/angular#52429]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

The `afterRender` hooks are called after each root view refreshes. This happens because `ApplicationRef.tick` calls `detectChanges` on each root view https://github.com/angular/angular/blob/bdd61c768a28b56c68634b99c036986499829f45/packages/core/src/application_ref.ts#L1057-L1059
and the hooks are called after running change detection on any view https://github.com/angular/angular/blob/bdd61c768a28b56c68634b99c036986499829f45/packages/core/src/render3/instructions/change_detection.ts#L70-L80

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-cnr3id

### Please provide the exception or error you saw

```true
`afterRender` is called after each root view is refreshed. `ApplicationRef.tick` should only run the render hooks one time, after all root views are refreshed.
```

**Comment by AndrewKushnir** [comment]
// cc @devknoll

**Comment by devknoll** [comment]
Thanks for raising this issue!

We could address this by having tick `begin()` and `end()` a render operation (like implemented in #52455). This is probably the "safest" option in the short term.

However, there's a more general question of whether being tied _directly_ to change detection timing is the right approach. For example, we also find that event handlers will cause it to run multiple times, e.g. once for `keydown`, again for `keyup`, etc. It's would be worth considering e.g. only scheduling a microtask during change detection, which would effectively batch and resolve this issue in both cases.

cc @kara @jelbourn @alxhub
